profiles/tests/test_profiles_graphql_api.py

```diff
@@ -1540,3 +1548,81 @@ def test_profile_node_exposes_key_for_federation_gateway(rf, anon_user_gql_clien
         'type ProfileNode implements Node  @key(fields: "id")'
         in executed["data"]["_service"]["sdl"]
     )
+
+
+def test_can_query_claimable_profile_with_token(rf, user_gql_client):
+    profile = ProfileFactory(user=None, first_name="John", last_name="Doe")
+    claim_token = ClaimTokenFactory(profile=profile)
+    request = rf.post("/graphql")
+    request.user = user_gql_client.user
+
+    t = Template(
+        """
+        {
+            claimableProfile(token: "${claimToken}") {
+                id
+                firstName
+                lastName
+            }        
+        }
+        """
+    )
+    query = t.substitute(claimToken=claim_token.token)
+    expected_data = {
+        "claimableProfile": {
+            "id": to_global_id(type="ProfileNode", id=profile.id),
+            "firstName": profile.first_name,
+            "lastName": profile.last_name,
+        }
+    }
+    executed = user_gql_client.execute(query, context=request)
+
+    assert "errors" not in executed
+    assert dict(executed["data"]) == expected_data
+
+
+def test_cannot_query_claimable_profile_with_user_already_attached(rf, user_gql_client):
+    profile = ProfileFactory()
+    claim_token = ClaimTokenFactory(
+        profile=profile, expires_at=timezone.now() - timedelta(days=1)
```

Comment:
I think you forgot to change some logic here 🙂 right now it's just copypasted from the next test.

Reply:
Actually the accidental copypaste was the `expires_at` argument for the `ClaimTokenFactory`. Otherwise it's correct. The difference is in `ProfileFactory` arguments.

Reply:
alrighty 👍